Makefile

```diff
@@ -1,33 +1,41 @@
+# variables that should not be overridden by the user
 VERSION = edge
-TAG = $(VERSION)
-PREFIX ?= nginx-kubernetes-gateway
-
-GIT_COMMIT = $(shell git rev-parse HEAD)
+GIT_COMMIT = $(shell git rev-parse HEAD || echo "unknown")
 DATE = $(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
 
-TARGET ?= local
-
-KIND_KUBE_CONFIG_FOLDER = $${HOME}/.kube/kind
-OUT_DIR=$(shell pwd)/build/.out
+# variables that can be overridden by the user
+PREFIX ?= nginx-kubernetes-gateway## The name of the image. For example, nginx-kubernetes-gateway
+TAG ?= $(VERSION:v%=%)## The tag of the image. For example, 0.3.0
+TARGET ?= local## The target of the build. Possible values: local and container
+KIND_KUBE_CONFIG_FOLDER = $${HOME}/.kube/kind## The folder where the kind kubeconfig is stored
+OUT_DIR ?= $(shell pwd)/build/.out## The folder where the binary will be stored
```

Comment:
Do we really need `OUT_DIR`? And why is it a hidden folder? Can we just save the binary in the root of the project?

Reply:
`build/out` dir is reasonable to me (without the hidden folder). it is also in .gitgnore